src/utils.ts

```diff
@@ -1,190 +1,185 @@
+  // find and return the path to the first encountered tsconfig.json file
+
+  if (!startDir) {
+    return getPathToClosestTSConfig(getStartDir());
+  }
+
+  const tsConfigPath = path.join(startDir, 'tsconfig.json');
+
+  const startDirPath = path.resolve(startDir);
+  const previousDirPath = path.resolve(previousDir || '/');
+
+  if (startDirPath === previousDirPath || fs.existsSync(tsConfigPath)) {
+    return tsConfigPath;
+  }
+
+  return getPathToClosestTSConfig(path.join(startDir, '..'), startDir);
+}
+
+function getTSConfigPathFromConfig(globals: ConfigGlobals): string {
+  const defaultTSConfigFile = getPathToClosestTSConfig();
+  if (!globals) {
+    return defaultTSConfigFile;
+  }
+
+  const tsJestConfig = getTSJestConfig(globals);
+
+  if (tsJestConfig.tsConfigFile) {
+    return tsJestConfig.tsConfigFile;
+  }
+
+  return defaultTSConfigFile;
+}
+
+export function mockGlobalTSConfigSchema(
+  globals: ConfigGlobals,
+): ConfigGlobals {
+  const configPath = getTSConfigPathFromConfig(globals);
+  return { 'ts-jest': { tsConfigFile: configPath } };
+}
+
+export const getTSConfig = _.memoize(getTSConfig_local, (globals, rootDir) => {
+  // check cache before resolving configuration
+  // NB: We use JSON.stringify() to create a consistent, unique signature. Although it lacks a uniform
+  //     shape, this is simpler and faster than using the crypto package to generate a hash signature.
+  return JSON.stringify(globals, rootDir);
+});
+
+// Non-memoized version of TSConfig
+function getTSConfig_local(globals, rootDir: string = '') {
+  const configPath = getTSConfigPathFromConfig(globals);
+  logOnce(`Reading tsconfig file from path ${configPath}`);
+  const skipBabel = getTSJestConfig(globals).skipBabel;
+
+  const config = readCompilerOptions(configPath, rootDir);
+  logOnce('Original typescript config before modifications: ', { ...config });
+
+  // ts-jest will map lines numbers properly if inlineSourceMap and
+  // inlineSources are set to true. The sourceMap configuration
+  // is used to send the sourcemap back to Jest
+  delete config.sourceMap;
+  config.inlineSourceMap = true;
+  config.inlineSources = true;
+
+  // the coverage report is broken if `.outDir` is set
+  // see https://github.com/kulshekhar/ts-jest/issues/201
+  // `.outDir` is removed even for test files as it affects with breakpoints
+  // see https://github.com/kulshekhar/ts-jest/issues/309
+  delete config.outDir;
+
+  if (configPath === path.join(getStartDir(), 'tsconfig.json')) {
+    // hardcode module to 'commonjs' in case the config is being loaded
+    // from the default tsconfig file. This is to ensure that coverage
+    // works well. If there's a need to override, it can be done using
+    // a custom tsconfig for testing
+    config.module = tsc.ModuleKind.CommonJS;
+  }
+
+  config.module = config.module || tsc.ModuleKind.CommonJS;
+  config.jsx = config.jsx || tsc.JsxEmit.React;
+

```

Comment:
I think this is the only real change. I have no idea what's going on with the diff.